doc/inlining.md

```diff
@@ -1,35 +1,61 @@
-## Warmup
+## Simple inlining
 
 As a simple warmup example lets consider inlining a function without osrs.
-Using some fresh variables for all variables of the function we simulate a letrec
-block to evaluate the body inside.
+
 
 ```
 function main()
   var x = 1
-  call y = 'foo(x)
+  call y = 'foo(x) -> l
   print y
 
 function foo(var x)
-  return x
+  var t = nil
+  read t
+  branch t $z $x
+  $z:
+    var z = x + x
+    print z
+    return z
+  $x:
+    return x
 ```
 
+(the `-> l` syntax indicates that `l` is the "return label" of the
```

Comment:
I thought you liked the idea of calls having a return label in their structure. Why not use it in the spec as well? I personally feel that using `goto (l+1)` for each `return` in the callee body is sort of inelegant. (If there was just one occurrence, fine, but we use it several time, I would rather have a proper name for the label.) If we cannot have `-> l`, I'll just consistently label the next expression `l` in all my examples.